lib/mayaUsd/ufe/UsdUndoAddNewPrimCommand.cpp

```diff
@@ -0,0 +1,100 @@
+//
+// Copyright 2019 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "UsdUndoAddNewPrimCommand.h"
+
+#include <mayaUsd/ufe/Global.h>
+#include <mayaUsd/ufe/Utils.h>
+
+namespace {
+
+Ufe::Path appendToPath(const Ufe::Path& path, const std::string& name)
+{
+    Ufe::Path newUfePath;
+    if (1 == path.getSegments().size()) {
+        newUfePath = path + Ufe::PathSegment(
+        Ufe::PathComponent(name), MayaUsd::ufe::getUsdRunTimeId(), '/');
+    } else {
+        newUfePath = path + name;
+    }
+    return newUfePath;
+}
+
+}
+
+MAYAUSD_NS_DEF {
+namespace ufe {
+
+UsdUndoAddNewPrimCommand::UsdUndoAddNewPrimCommand(const UsdSceneItem::Ptr& usdSceneItem,
+                            const std::string& name, const std::string& type)
+: Ufe::UndoableCommand()
+{
+    // First get the stage from the proxy shape.
+    auto ufePath = usdSceneItem->path();
+    auto segments = ufePath.getSegments();
+    auto dagSegment = segments[0];
+    _stage = MayaUsd::ufe::getStage(Ufe::Path(dagSegment));

```

Comment:
Since you have a UsdSceneItem, to get the stage you can just use: usdSceneItem->prim().GetStage()